### PR TITLE
feat(typography): add new Span component page

### PR DESCRIPTION
**Pull request by jzempel**
## Description

While the Span page is the main feature of this PR, I refactored the "Typography" section in order to give components dedicated page treatment. Fixed a few bugs along the way and added examples to support features of the various components.

@allisonacs, @rossmoody an IA critique is in order. 

@vsrixyz the entire section will need a thorough look as I made scores of solo copy calls along the way.

## Checklist

- [x] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [x] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :zap: audited via [web.dev](https://web.dev/measure/) for performance and SEO
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/pages/components/span.mdx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+components:
+  - typography/src/elements/MD.tsx
+  - typography/src/elements/Paragraph.tsx
+  - typography/src/elements/Span.tsx
+---
+
+<Usage>
+  <Use>
+    <ul>
+      <li>To apply inline styles</li>
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:15]: - To apply styles to short strings of text

File: src/pages/components/span.mdx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      <li>To include icons inline with text</li>
+    </ul>
+  </Use>
+  <Misuse>
+    <ul>
+      <li>
+        <Markdown>As a replacement for a semantic [paragraph](/components/paragraph)</Markdown>
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:22]: To style an entire block of text, use Paragraph instead

File: src/pages/components/span.mdx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      </li>
+    </ul>
+  </Misuse>
+</Usage>
+
+## How to use it
+
+### Bold
+
+`isBold` applies a bold typography style without conveying any extra emphasis or importance.
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:32]: To apply bold styling without indicating semantic importance use `isBold`.

File: src/pages/components/span.mdx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import Bold from '../../examples/typography/span/Bold.tsx';
+import BoldCode from '!!raw-loader!../../examples/typography/span/Bold.tsx';
+
+<CodeExample code={BoldCode}>
+  <Bold />
+</CodeExample>
+
+### Hue
+
+You can change the color of the span by specifying a different `hue`.
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:43]: The color of a span is specified with `hue`.

File: src/pages/components/span.mdx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+import Hue from '../../examples/typography/span/Hue.tsx';
+import HueCode from '!!raw-loader!../../examples/typography/span/Hue.tsx';
+
+<CodeExample code={HueCode}>
+  <Hue />
+</CodeExample>
+
+### Icons
+
+You can add icons to Spans.
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:54]: ~You can add icons to Spans~

File: src/pages/components/span.mdx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+To include an icon at the beginning of a span, use `<Span.StartIcon>`.
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:56]: To include an icon at the beginning of a span, use `<Span.StartIcon>`.

File: src/pages/components/span.mdx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+import StartIcon from '../../examples/typography/span/StartIcon.tsx';
+import StartIconCode from '!!raw-loader!../../examples/typography/span/StartIcon.tsx';
+
+<CodeExample code={StartIconCode}>
+  <StartIcon />
+</CodeExample>
+
+Use `<Span.Icon>` to add icons in the middle or at the end of a Span.
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:65]: To include an icon anywhere else in the Span use `<Span.Icon>`.

File: src/pages/components/span.mdx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+import Icon from '../../examples/typography/span/Icon.tsx';
+import IconCode from '!!raw-loader!../../examples/typography/span/Icon.tsx';
+
+<CodeExample code={IconCode}>
+  <Icon />
+</CodeExample>
+
+### Monospace
+
+Use `isMonospace` to apply a monospaced type treatment.
```
Comment (vsrixyz) [inline src/pages/components/span.mdx:76]: Apply a fixed-width type treatment using `isMonospace`.

File: src/pages/components/paragraph.mdx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+import { graphql } from 'gatsby';
+
+export const pageQuery = graphql`
+  query($fileAbsolutePath: String) {
+    ...SidebarPageFragment
+  }
+`;
+
+;
```
Comment (hzhu) [inline src/pages/components/paragraph.mdx:69]: I think this might be an extra semicolon.

File: src/pages/components/span.mdx [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+      </li>
+    </ul>
+  </Misuse>
+</Usage>
+
+## How to use it
+
+### Bold
+
+To apply bold styling without indicating semantic importance use `isBold`.
```
Comment (allisonacs) [inline src/pages/components/span.mdx:34]: Add a comma between "importance" and "use"

File: src/pages/components/span.mdx [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+To include an icon at the beginning of a span, use `<Span.StartIcon>`.
+
+import StartIcon from '../../examples/typography/span/StartIcon.tsx';
+import StartIconCode from '!!raw-loader!../../examples/typography/span/StartIcon.tsx';
+
+<CodeExample code={StartIconCode}>
+  <StartIcon />
+</CodeExample>
+
+To include an icon anywhere else in the Span use `<Span.Icon>`.
```
Comment (allisonacs) [inline src/pages/components/span.mdx:65]: Add a comma between "Span" and "use"

**Comment by allisonacs** [comment]
And a note/to-do. When we have a `code block` in addition to `code`, we will need to differentiate when to use which.

File: src/pages/components/paragraph.mdx [hunk 12]
```diff
@@ -0,0 +1,67 @@
+---
+title: Paragraph
+
+description: Use Paragraph to render text blocks with the appropriate Garden styling.
```
Comment (vsrixyz) [inline src/pages/components/paragraph.mdx:4]: Let's change to: 

Use Paragraph to render blocks of text with Garden styling.

File: src/pages/components/paragraph.mdx [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+import Default from '../../examples/typography/paragraph/Default.tsx';
+import DefaultCode from '!!raw-loader!../../examples/typography/paragraph/Default.tsx';
+
+<CodeExample code={DefaultCode}>
+  <Default />
+</CodeExample>
+
+### Size
+
+Paragraphs come in small, medium, and large. The default size is medium. Size
```
Comment (vsrixyz) [inline src/pages/components/paragraph.mdx:28]: Size controls the space between Paragraphs. The default size is medium, and you can choose small or large too. Use [typography](/components/typography) to choose an appropriate type scale.

**Pull request merged**